src/clims/api/serializers/models/substance.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import absolute_import
+
+from clims.models.substance import Substance
+from rest_framework import serializers
+from rest_framework.fields import Field
+
+
+class PropertyValueField(Field):
+    def to_representation(self, obj):
+        return obj
+
+    def to_internal_value(self, data):
+        return data
+
+
+class PropertyTypeSerializer(serializers.Serializer):
+    id = serializers.IntegerField(read_only=True)
+
+
+class PropertyOverviewSerializer(serializers.Serializer):
+    id = serializers.IntegerField(read_only=True)
+    type = PropertyTypeSerializer()
+    value = PropertyValueField()
+
+
+class PropertiesField(Field):
+    def to_representation(self, obj):
+        ret = dict()
+        for prop in obj.filter(latest=True):
+            display_name = prop.type.display_name if prop.type.display_name else prop.type.name
+            ret[prop.type.name] = dict(value=prop.value, display_name=display_name)
+        return ret
+
+    def to_internal_value(self, data):
+        return data
+
+
+class VersionField(Field):
+    def to_representation(self, obj):
+        return obj
+
+
+class ExtensibleTypeField(Field):
+    def to_representation(self, obj):
+        return "{}.substances.{}".format(obj.plugin.name, obj.name)
+
+    def to_internal_value(self, data):
+        return data
+
+
+class SubstanceSerializer(serializers.Serializer):
+
```

Comment:
I'm not sure I understand these serializers fully. How is it with the foreign key navigation properties that exists within substance, like project, and the many-to-many navigation properties, origins and parents are not here, but properties are. Shouldn't depth be here?

Reply:
In the serialization "layer" one can decide what the REST API users will see. One argument is that they should see everything, and that makes a lot of sense in that it's simple, but it also gives access to data that the frontend might not need.

My thinking was to defer that decision to later and only start with what we know the frontend requires. But I would also be fine with us just serializing everything directly - so please add what you need too :)

The rest_framework has serializers that serialize whole models and make this easier, but in this case I wrote our own because the substance relates to all those different models to do what it needs to do so I assumed it would be simpler to just throw together a serializer.

Reply:
Ok, thats clear to me now.